wp-content/mu-plugins/register-rest-routes.php

```diff
@@ -288,3 +290,141 @@
     }
   ));
 });
+
+add_action('rest_api_init', function() {
+  $v = 'api/v1'; // namespace for the current version of the API
+
+  $exp = WEEK_IN_SECONDS; // expiration of the transient caches  
+
+  function LogEventData($eventData){
+    $log_dir = WP_CONTENT_DIR . '/sendgrid-events';
+    $log_file = $log_dir . '/events.log'; 
+    $event_type = isset($eventData['event']) ? $eventData['event'] : 'unknown';
+    $templateName = isset($eventData['sg_template_name']) ? $eventData['sg_template_name'] : 'unknown';
+    $email = isset($eventData['email']) ? $eventData['email'] : 'unknown';
+    $timestamp = isset($eventData['timestamp']) ? $eventData['timestamp'] : 'unknown';
+    $url = isset($eventData['url']) ? $eventData['url'] : 'unknown';
+    $log_data = "Event: $event_type | URL: $url | Timestamp: $timestamp\n";
+    file_put_contents($log_file, $log_data, FILE_APPEND);
+    return array('event'=>$event_type, 'email'=>$email, 'templateName'=>$templateName);
+  }
+  
+  register_rest_route('api/v1', '/newsletter/confirm/', array(
+    'methods' => 'POST',
+    'permission_callback' => '__return_true',
+
+    'callback' => function(WP_REST_Request $request) {
+      $input = $request->get_body();
+      $data = json_decode($input, true);
+
+      if (is_array($data)) {     
+        foreach ($data as $event) {
+          $event_data = LogEventData($event);          
+          if($event_data['event']=='click' && $event_data['templateName']==SENDGRID_CONFIRMED_TEMPLATE_NAME){
+            $apiKey = SENDGRID_API_KEY;
+            $sg = new \SendGrid($apiKey);
+            $request_body = json_decode('{
```

Comment:
- [ ] Create a PHP array directly instead of writing a JSON and converting it to a PHP array